src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -221,13 +222,15 @@ export const handleEmailSubmission: RequestHandler<
     submissionId: submission.id,
   })
 
+  const autoReplyData = new autoReplyDataObj(emailData.autoReplyData)
```

Comment:
there doesn't seem to be a need to access the raw `autoReplyData` object anywhere. might it be better to call this constructor in `EmailSubmissionService.createEmailData` so the data is encapsulated from the moment it is created?

in general, we don't want the controllers to have to worry about implementation details such as calling a constructor on a raw data object. ideally the interfaces of the services should be set up seamlessly so the controller can easily pass data between them.

Reply:
edited - I've also changed the class to encapsulate all of EmailData, which I think is neater. So in future we can define other methods to modify / retrieve the other elements of EmailData within the same encapsulation, if necessary.